content/post/articles/how-to-build-an-oss-project.md

```diff
@@ -0,0 +1,139 @@
+
+You could then expect the home page to be ready in a month, but as we often observe 99% of the time, this won’t be true. But at least you might have something close to a fully functional home page.
+
+#### The Architecture
+
+Regarding the architecture of your project, the keyword here is **simplicity**. As developers, we often like to work on complicated systems, implement challenging algorithms and use the new fancy DevOps practices or tools (Kubernetes) we saw on Twitter. These are all noble things to thrive for, but they come with a cost.
+
+When we start a new open source project, we are excited about the problem we want to solve, the challenges we want to face, and what we will learn. This keeps us motivated long enough to release the project, but users start submitting feature requests or pointing out bugs as long as the project is out there. These incentives are not sufficient anymore, and the nice and fun development becomes a long-term maintenance chore.
+
+Building applications made of multiple components in a microservice architecture with a database or a Saas solution comes with a few operational challenges and costs that should not be overlooked. The way you architecture or write your code and the technologies you choose might also harm the long-term maintenance of your project.
+
+Maintaining an open-source project is a long-term commitment. And keeping things simple will considerably improve first-time contributions and your motivation to maintain your projects.
+
+**How to keep things simple:**
+
+- **Build a monolith** by default. Sometimes you might want to split your frontend and backend to ease the development of your team. It is totally fine. But build a monolith as far as the backend is concerned.
+
+- **Mono repositories** are crucial to keep your application code at hand. It is easier to know where things happen when they happen in one place.
+
+- **Avoid Saas.** Using subscriptions Saas like [Firebase][3], [MongoDB Atlas][4], or something not already used by the community can make operational work difficult. If you absolutely think you need a saas subscription [contact us][5], we will probably try to convince you that you don’t, but we are open to changing our mind or getting it for you.
+
+- **Avoid self-managed Databases** as they require a massive amount of operational work. Generally speaking, we recommend you design your application using file **databases** like a YAML file or SQL-lite. [camerapps][6] is an excellent example of an application that uses a YAML file as a Database. We are working on making more use of SQL lite.
+
+- **Ask for a managed Database** we acknowledge that it might be essential to spin up a database for a few projects. If you think your application needs one [talk to us][7], we probably think it doesn’t, but we are open.
```

Comment:
```suggestion
- **Ask for a managed Database** we acknowledge that it might be essential to spin up a database for a few projects. If you think your application needs one [talk to us][7], we think it probably doesn’t, but we are open.
```